browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -115,15 +109,28 @@ void OnShouldBlockAdResult(const ResponseCallback& next_callback,
   next_callback.Run();
 }
 
-void ShouldBlockAdWithOptionalCname(
-    scoped_refptr<base::SequencedTaskRunner> task_runner,
-    const ResponseCallback& next_callback,
-    std::shared_ptr<BraveRequestInfo> ctx,
-    const base::Optional<std::string> cname) {
+void UseCnameResult(scoped_refptr<base::SequencedTaskRunner> task_runner,
+                    const ResponseCallback& next_callback,
+                    std::shared_ptr<BraveRequestInfo> ctx,
+                    const base::Optional<std::string> cname) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-  task_runner->PostTaskAndReply(
-      FROM_HERE, base::BindOnce(&ShouldBlockAdOnTaskRunner, ctx, cname),
-      base::BindOnce(&OnShouldBlockAdResult, next_callback, ctx));
+
+  if (cname.has_value() && ctx->request_url.host() != *cname &&
+      !cname->empty()) {
+    GURL::Replacements replacements = GURL::Replacements();
```

Comment:
I think you can just declare it without setting it equal to the static method per the following

https://source.chromium.org/chromium/chromium/src/+/main:url/url_canon.h;l=762;drc=1d94f5d9a4c74aa787b9b27fca44a4ee99869ef6

Reply:
fixed